src/Exception/DiscoveryFailedException.php

```diff
@@ -12,22 +12,37 @@
 final class DiscoveryFailedException extends \Exception implements Exception
 {
     /**
-     * @var array
+     * @var \Exception[]
      */
     private $exceptions;
 
     /**
-     * @param $exceptions
+     * @param string       $message
+     * @param \Exception[] $exceptions
      */
     public function __construct($message, array $exceptions = [])
     {
         $this->exceptions = $exceptions;
 
-        parent::__construct($message, 0, array_shift($exceptions));
+        parent::__construct($message);
```

Comment:
this is eating away the chance to see a root cause exception. in general, it could be useful to see where the original problem comes from. but in the situation i agree it would be more confusing than helpful.